app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -43,22 +87,32 @@ public function createOrder($order)
 
         $currency = Razorpay_Payments_Model_Paymentmethod::CURRENCY;
 
-        $url = $this->getRelativeUrl('order');
+        $razorpayOrderId = Mage::getSingleton('core/session')->getRazorpayOrderID();
 
-        $data = array(
-            'receipt'         => $orderId,
-            'amount'          => $amount,
-            'currency'        => $currency,
-            'payment_capture' => 1
-        );
+        if ($razorpayOrderId === null)
+        {
+            Mage::log(['razorpayOrderId' => 'NULL']);
```

Comment:
It would be good in debugging. In the case when double orders are created, if razorpayOrderId is not logged as null, it would be indicative of the error.